addok_fr/utils.py

```diff
@@ -33,13 +36,15 @@ def phonemicize(s):
             ("(?<=u)lt$", "t"),
             ("(?<=[a-z])[dg]$", ""),
             ("(?<=[^es0-9])t$", ""),
-            ("(?<=[aeiou])(m)(?=[pbgf])", "n"),
+            ("(?<=[aeiou])(m)(?=[pbgft])", "n"),
             ("(?<=[a-z]{2})(e$)", ""),  # Remove "e" at last position only if
                                         # it follows two letters?
-            ("(\\D)(?=\\1)", ""),  # Remove duplicate letters.
+            ("([aeiouy])n[dt]([^aeiouyr])", "\\1n\\2"),  # montbon -> monbon
+            (r"([a-z])\1", r"\1"),  # Remove duplicate letters.
```

Comment:
This one is not gready, if someone types three "t", it will only remove one of those.
And it seems slower with a double letter, but I've just made a quick check:
1000000 loops, best of 3: 1 µs per loop vs 100000 loops, best of 3: 2.7 µs per loop